File: prob8/prob8.rb
class Discount

  def initialize(total,percent)
    @total     = total
    @percent   = percent
  end


	def calc_discount()
		if @percent > 100 or @percent < 0
			return "Sorry. Enter a valid percent."
		else
			the_discount = @total * (@percent * 0.01)
			return the_discount.round(2)
		end
	end

end


# discount1 = Discount.new(100,10)
# puts discount1.calc_discount()

# discount2 = Discount.new(90,90)
# puts discount2.calc_discount()

# discount3 = Discount.new(87.9,91.6)
# puts discount3.calc_discount()

# discount4 = Discount.new(90,-1)
# puts discount4.calc_discount()

File: problems/prob08/prob8.py
class Discount:
	@staticmethod
	def calc_discount(total, percent):
		if percent > 100 or percent < 0:
			return "Sorry. Enter a valid percent."
		else:
			the_discount = total * (percent * .01)
			return round(the_discount,2)



# discount1 = Discount()
# print discount1.calc_discount(100,10)
# print discount1.calc_discount(90,90)
# print discount1.calc_discount(87.9,91.6)
# print discount1.calc_discount(90,-1)